src/include/storage/index/hash_index_builder.h

```diff
@@ -62,13 +69,21 @@ class HashIndexBuilder final : public InMemHashIndex {
     // Reserves space for at least the specified number of elements.
     void bulkReserve(uint32_t numEntries) override;
 
-    bool append(T key, common::offset_t value);
-    bool lookup(T key, common::offset_t& result);
+    using AppendKeyType =
```

Comment:
Maybe `BufferKeyType` is more accurate here? It looks to me that the type is more about the type info of stored keys inside the IndexBuffer.